coding_systems/dmd/coding_system.py

```diff
@@ -37,3 +37,36 @@ def code_to_term(self, codes):
         lookup = self.lookup_names(codes)
         unknown = set(codes) - set(lookup)
         return {**lookup, **{code: "Unknown" for code in unknown}}
+
+    def codes_by_type(self, codes, hierarchy):
+        """
+        The entities we search across are all different "types"
+        but they share a common ancestor of a medicinal product.
+        Since we build them all into a singular hierarchy,
+        displaying them in separate "type" sections doesn't make
+        sense.
+        """
+        codes = (
+            list(
+                Ing.objects.using(self.database_alias)
+                .filter(id__in=codes)
+                .values_list("id", flat=True)
+            )
+            + list(
+                VTM.objects.using(self.database_alias)
+                .filter(id__in=codes)
+                .values_list("id", flat=True)
+            )
+            + list(
+                VMP.objects.using(self.database_alias)
+                .filter(id__in=codes)
+                .values_list("id", flat=True)
+            )
+            + list(
+                AMP.objects.using(self.database_alias)
+                .filter(id__in=codes)
+                .values_list("id", flat=True)
+            )
+        )
+
```

Comment:
erm - can you just return `{"Product": codes}` without all the db calls? I suppose this is slightly different in that if any of the `codes` are incorrect then this filters them out - but we don't do that for other coding systems. Am I missing something?

Reply:
AFAICT the other coding systems put "unknown" codes in their own "unknown" category. We should probably do that here.